### PR TITLE
Implementation of the GetProof methods

### PR DESCRIPTION
**Pull request by novosandara**
# Description

Implementation of the GetProof methods

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [x] I have tested this code manually

### CODE REVIEW

File: jsonrpc/eth_endpoint.go [hunk 1]
```diff
@@ -1087,3 +1089,89 @@ func (e *Eth) signTx(args *txnArgs) (*types.Transaction, error) {
...
+	}
+
+	var codeHash types.Hash
+
+	nonce := uint64(0)
+
+	storageHash := snap.GetTreeHash()
+
+	var balance *argBig
+	balance = nil
```
Comment (Stefan-Ethernal) [inline jsonrpc/eth_endpoint.go]:
```suggestion
```

File: jsonrpc/eth_endpoint.go [hunk 2]
```diff
@@ -1087,3 +1089,89 @@ func (e *Eth) signTx(args *txnArgs) (*types.Transaction, error) {
...
+	if err != nil {
+		return nil, err
+	}
+
+	var codeHash types.Hash
+
+	nonce := uint64(0)
+
+	storageHash := snap.GetTreeHash()
+
```
Comment (Stefan-Ethernal) [inline jsonrpc/eth_endpoint.go]: Nitpick, group all variables within the var block:

```suggestion
	var (
		codeHash types.Hash
		nonce = uint64(0)
		storageHash = snap.GetTreeHash()
		balance *argBig
	)
```

File: jsonrpc/eth_endpoint.go [hunk 3]
```diff
@@ -1087,3 +1089,89 @@ func (e *Eth) signTx(args *txnArgs) (*types.Transaction, error) {
...
+	if err != nil {
+		return nil, err
+	}
+
+	snap, err := e.store.NewSnapshotAt(header.StateRoot)
+	if err != nil {
+		return nil, err
+	}
+
+	account, err := snap.GetAccount(address)
```
Comment (Stefan-Ethernal) [inline jsonrpc/eth_endpoint.go]: Should we return an error if there is no account found?

**Comment by oliverbundalo** [comment]
Closing this PR, we will keep the branch with implementation.

**Pull request closed without merging**